crates/iceberg/src/scan/mod.rs

```diff
@@ -280,6 +315,7 @@ impl<'a> TableScanBuilder<'a> {
             partition_filter_cache: Arc::new(PartitionFilterCache::new()),
             manifest_evaluator_cache: Arc::new(ManifestEvaluatorCache::new()),
             expression_evaluator_cache: Arc::new(ExpressionEvaluatorCache::new()),
+            file_column_position,
```

Comment:
Having a separate field for file column position is not ideal, because it doesn't scale, i.e. what will we do when we need to add more virtual/metadata columns and their positions.
So perhaps preserving them through field IDs vector would inherently preserve positions?

Reply:
Changed that